cpp/src/copying/copy_range.cu

```diff
@@ -42,21 +44,20 @@ struct inplace_copy_range_dispatch {
   std::enable_if_t<cudf::is_fixed_width<T>(), void>
   operator()(cudf::size_type source_begin, cudf::size_type source_end,
              cudf::size_type target_begin, cudaStream_t stream = 0) {
-    if (target.nullable()) {
-      auto p_device_view =
-        cudf::column_device_view::create(source, stream);
+    auto p_source_device_view =
+      cudf::column_device_view::create(source, stream);
+    if (p_source_device_view->has_nulls()) {
       cudf::experimental::detail::copy_range(
-        source.begin<T>() + source_begin,
-        cudf::experimental::detail::make_validity_iterator(*p_device_view) +
-          source_begin,
+        cudf::experimental::detail::make_null_replacement_iterator<T>(
```

Comment:
I don't think this needs a null_replacement_iterator. That just adds extra overhead. It's okay to return garbage values for the null elements by just using `.begin<T>()`.

Reply:
I initially implemented this with .begin<T>(), but then, I encountered

https://github.com/rapidsai/cudf/blob/885ac9bbd61c460cdb6f61c1ad1258b711da425d/cpp/include/cudf/column/column_device_view.cuh#L637

column_view's .begin<T> just returns a pointer so this wasn't an issue (and no specialization for strings) but column_device_view's begin returns an iterator using value_accessor.

https://github.com/rapidsai/cudf/blob/885ac9bbd61c460cdb6f61c1ad1258b711da425d/cpp/include/cudf/column/column_device_view.cuh#L288

That's why I used null_replacement_iterator, but if you have a better solution, please let me know.

Reply:
And it seems like

https://github.com/rapidsai/cudf/blob/885ac9bbd61c460cdb6f61c1ad1258b711da425d/cpp/include/cudf/column/column_device_view.cuh#L637

should be !_col.nullable() (instead of !_col.has_nulls()) to match the comment

https://github.com/rapidsai/cudf/blob/885ac9bbd61c460cdb6f61c1ad1258b711da425d/cpp/include/cudf/column/column_device_view.cuh#L613

(or the comment should be updated).